subprojects/dependency-management/src/main/java/org/gradle/internal/locking/LockFileReaderWriter.java

```diff
@@ -24,22 +25,29 @@
 
 import java.io.IOException;
 import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.Predicate;
 
 public class LockFileReaderWriter {
 
     private static final Logger LOGGER = Logging.getLogger(LockFileReaderWriter.class);
     private static final DocumentationRegistry DOC_REG = new DocumentationRegistry();
 
+    static final String UNIQUE_LOCKFILE_NAME = "all_locks.singlelockfile";
```

Comment:
:o: suggestions:
- `project.lockfile`
- `<rootdirectory>/dependency-locking/<project-name>.lockfile` (I like this one because all lock files live in the same directory instead of distributed accross all projects)

Reply:
I picked the first one, easier to do. And I kind of like the per project aspect. Tt shows the scope better IMO.

Reply:
However it creates a double `gradle/dependency-locks` directory in each project, which kind of bloats the output, I feel. We also have the verification metadata which lives into `<root>/gradle`, so I'm a bit worried about having files exploded into several directories. That said, the convention `<project-name>.lockfile` is not enough (probably needs to be project _path_ or a transformation of the project path).